ethereum/casper.se.py

```diff
@@ -361,6 +361,10 @@ def submitBet(index:uint256, max_height:uint256, probs:bytes, blockhashes:bytes3
 def const sampleGuardian(orig_seedhash:bytes32, blknumber:uint256):
     n = self.nextGuardianIndex
     seedhash = sha3([orig_seedhash, blknumber]:arr)
+
+    while n > 1 and self.guardians[n - 1].induction_height > blknumber:
+        n -= 1
+
```

Comment:
@vbuterin can you let me know what you think about this change?

When I am running guardian nodes separately, they would end up disagreeing on who should produce the next block once a new guardian would join.  Prior to this change two independent queries about who should produce a given block could return different values depending on whether their `state` included the new guardian node.  I *think* that this change ensures that calls to `sampleGuardian` always return the same value for a given block number.